functiondefextractor/core_extractor.py

```diff
@@ -26,47 +26,151 @@ def get_file_names(dir_path):
-            file_content = file_data.readlines()
-            iterator = int(line_num) - 2
-            # print(iterator)
-            for _ in range(int(line_num) - 2):
-                data = str(file_content[iterator]).strip().upper()
-                iterator = iterator - 1
-                if annot.upper() in data.upper():  # Change to start with
-                    code = get_func_body(filename, line_num)
-                    return code
-                elif data[:1] is not "@" and "}" in data or "{" in data:
-                    return None
+        return get_annot_methods(filename, line_num, annot)
+
+
+def get_file_content(filename):
+    """ Function reads the given file
+        @parameters
+        filename: Path to the file
+        @return
+        This function returns content of the file inputed"""
+    with open(filename) as file_data:
+        return file_data.readlines()
+
+
+def get_annot_methods(filename, line_num, annot):
+    """ This function gets the methods that have given annotation
+        @parameters
+        filename: Path to the file
+        line_num: function/method line number
+        annot: Annotation condition (Ex: @Test)
+        @return
+        This function returns function/method definitions that has the given annotation"""
+    file_content = get_file_content(filename)
+    iterator = int(line_num) - 2
+    for _ in range(int(line_num) - 2):
+        data = str(file_content[iterator]).strip().upper()
+        iterator = iterator - 1
+        ret_val = process_annot_method_body(annot, data, filename, line_num)
+        if ret_val != "continue":
+            return ret_val
+
+
+def process_annot_method_body(annot, data, filename, line_num):
+    """ This function process the function/method to check if it has the given annotation
+        @parameters
+        filename: Path to the file
+        line_num: function/method line number
+        annot: Annotation condition (Ex: @Test)
+        data: Content of the given file
+        @return
+        This function returns function/method definitions that has the given annotation"""
+    ret_val = "continue"
+    if annot.upper() in data.upper():
+        ret_val = get_func_body(filename, line_num)
+    elif data[:1] is not "@" and "}" in data or "{" in data:
+        ret_val = None
+    return ret_val
+
+
+def check_py_annot(file_name, annot):
+    """ Function checks for the annotation condition in python files
+        @parameters
+        filename: Path to the file
+        annot: Annotation condition (Ex: @Test)
+        @return
+        This function returns function/method names that has the given annotation"""
+    line_data = list([line.rstrip() for line in open(file_name)])
+    data = []
+    val = 0
+    if annot == "test_":
+        annot = "def test_"
+        val = -1
+    for i, _ in enumerate(line_data):
+        if annot in line_data[i]:
+            func_name = line_data[i + 1 + val].strip().split(" ")[1].split("(")[0]
```

Comment:
While splitting there could be chance that the result is not as expected. There could be exception here!